integration-tests/src/tests/network/runner.rs

```diff
@@ -660,15 +552,31 @@ impl Runner {
         network_config.minimum_outbound_peers =
             test_config.minimum_outbound_peers.unwrap_or(network_config.minimum_outbound_peers);
 
-        setup_network_node(
-            accounts_id[node_id].clone(),
-            self.validators.clone().unwrap(),
-            self.genesis_time.unwrap(),
-            network_config,
-        )
+        let (send_pm, recv_pm) = tokio::sync::oneshot::channel();
+        let (send_stop, recv_stop) = tokio::sync::oneshot::channel();
+        let handle = std::thread::spawn({
+            let account_id = accounts_id[node_id].clone();
+            let validators = self.validators.clone().unwrap();
+            let genesis_time = self.genesis_time.unwrap();
+            move || {
+                actix::System::new().block_on(async move {
+                    send_pm
+                        .send(setup_network_node(
+                            account_id,
+                            validators,
+                            genesis_time,
+                            network_config,
+                        ))
+                        .map_err(|_| anyhow!("send failed"))?;
+                    recv_stop.await?;
+                    Ok(())
+                })
+            }
+        });
+        Ok(NodeHandle { addr: recv_pm.await?, send_stop: send_stop, handle: handle })
```

Comment:
```suggestion
        Ok(NodeHandle { addr: recv_pm.await?, send_stop, handle })
```

Reply:
Also, ultraminor, but consider `let addr = recv_pm.await?` as well. It seem preferable to move side-effects out of struct literal contruction.

Reply:
done